Client side app routing

valvegan opened the pull request.

valvegan committed:
initial

valvegan committed:
Update README.md

valvegan committed:
Update README.md

valvegan committed:
reinstallation of parcel, correct version

valvegan committed:
local parcel installation

valvegan committed:
backbutton to movieview

valvegan committed:
Update README.md

valvegan committed:
Update README.md

valvegan committed:
Update README.md

valvegan committed:
mockup movie descriptions

valvegan committed:
axios installation

valvegan committed:
axios - componentDidMount()

valvegan committed:
registration

valvegan committed:
proptypes for movies

valvegan committed:
login and registration components

- proptypes added for login, registration, and movie-card components

valvegan committed:
comment out

conditional to check if the death entry is null - commented out

valvegan committed:
bootstrap react

valvegan committed:
grid layout

valvegan committed:
proptypes fixes and conditional rendering

update from fixes on the previous exercise

valvegan committed:
Update registration-view.jsx

valvegan committed:
registration form

valvegan committed:
Revert "Update registration-view.jsx"

This reverts commit 3c0c1e7f87133dad0d7ae066d71a0d021230efe7.

valvegan committed:
forms

valvegan committed:
small updates

need to resolve issue with styling !!!

valvegan committed:
edits

valvegan committed:
update

valvegan committed:
some updates

valvegan committed:
axios getMovies function

valvegan committed:
user auth local storage

valvegan committed:
log in view form validation!

valvegan committed:
spread syntax

valvegan committed:
registration view form validation

valvegan committed:
formatting

valvegan committed:
fixes with state routing

valvegan committed:
fixes on registration form

spread syntax not working!

valvegan committed:
movie actors fixes

spitired away has no actors so added the conditional

valvegan committed:
issue with fetching actor details

valvegan committed:
profile view edits

need to update the index.js file on the movie api project because i forgot to add a get user endpoint

valvegan committed:
profile form edits

valvegan committed:
not sure

valvegan committed:
Revert "not sure"

This reverts commit f5aa4c307d961d929479491ad5ae4357402135db.

valvegan committed:
fixes to actor view

valvegan committed:
fixed actors view

valvegan committed:
add to favorites and remove

valvegan committed:
post not put

valvegan committed:
edits to profile page

valvegan committed:
fix

valvegan committed:
formatted

Codilot commented on src/components/profile-view/profile-view.jsx:
```diff
@@ -0,0 +1,367 @@
+  FormControl,
+  Button,
+} from "react-bootstrap";
+import { throws } from "assert";
```
this import is throwing an error when I try to build your app

Codilot commented on src/components/actor-view/actor-view.jsx:
```diff
@@ -0,0 +1,62 @@
+import { match } from "assert";
```
This import is throwing an error when I build your app

Codilot commented on src/components/movie-view/movie-view.jsx:
```diff
@@ -0,0 +1,187 @@
+          `https://my-flix-api-2022.herokuapp.com/users/${user}/favoriteMovies/${id}`,
+
+          { headers: { Authorization: `Bearer ${token}` } }
+        )
```
The axios post request needs 3 parameter. It currently sends your header as the request body. instead you can add an empty object as the second parameter:
```
.post(
          `https://my-flix-api-2022.herokuapp.com/users/${user}/favoriteMovies/${id}`,
          {},
          { headers: { Authorization: `Bearer ${token}` } }
        )
```

Codilot commented on src/components/movie-view/movie-view.jsx:
```diff
@@ -0,0 +1,187 @@
+        `https://my-flix-api-2022.herokuapp.com/users/${user}/movies/${id}`,
+
+        { headers: { Authorization: `Bearer ${token}` } }
+      )
```
The axios delete request needs 3 parameter. It currently sends your header as the request body. instead you can add an empty object as the second parameter.

valvegan committed:
backgrounds and images update

just color changes and clickable movie-card images

valvegan committed:
add favorite first fixes

need to add conditional to only show "remove fav" button if movie has already been favorited
need to fix the remove favorite button on profile view

valvegan committed:
issue with delete request

valvegan committed:
fixed delete button

empty parameter goes after the authorization token

valvegan committed:
cancedl

valvegan committed:
Revert "cancedl"

This reverts commit a3656256eeffe14175e51445ef66624a6807e35f.

valvegan committed:
movie card fav button fixes

added conditional statements to add and remove favorite movies
copy pasted the function to get data about the user (its state) in order to check if the movie id is in the list of movies

valvegan committed:
fixed remove button on profile

valvegan committed:
format documents

indentations etc

valvegan committed:
fix

valvegan committed:
redux and react redux installation dependencies
